src/zip.h

```diff
@@ -192,6 +192,20 @@ extern int zip_entry_write(struct zip_t *zip, const void *buf, size_t bufsize);
 */
 extern int zip_entry_fwrite(struct zip_t *zip, const char *filename);
 
+/*
+  Compresses an input buffer for the current zip entry.
+
+  Args:
+    zip: zip archive handler.
+    buf: input buffer.
+    bufsize: input buffer size (in bytes).
+    sd_mode: file open mode. 0664 means user read/write, group read/write, all read.
+
+  Returns:
+    The return code - 0 on success, negative number (< 0) on error.
+*/
+extern int zip_entry_dwrite(struct zip_t *zip, const void *buf, size_t bufsize, int st_mode);
```

Comment:
OK, let's rename the function to sth. like: `zip_entry_writemode` or `zip_entry_writewithmode`.

Reply:
Also take a look how it was done for files:
```c
if ((file_stat.st_mode & 0200) == 0) {
    // MS-DOS read-only attribute
    zip->entry.external_attr |= 0x01;
  }
  zip->entry.external_attr |= (mz_uint32)((file_stat.st_mode & 0xFFFF) << 16);
```